src/Stratis.Bitcoin.Features.MemoryPool/MempoolValidator.cs

```diff
@@ -197,6 +196,13 @@ public async Task<bool> AcceptToMemoryPoolWithTime(MempoolValidationState state,
                 //    BOOST_FOREACH(const uint256& hashTx, vHashTxToUncache)
                 //        pcoinsTip->Uncache(hashTx);
                 //}
+
+                if (state.IsInvalid)
+                {
+                    this.logger.LogTrace("(-):false");
```

Comment:
Bellow (in the exception) we trace some additional stuff on failure maybe also add the same here?

Reply:
Do you mean we also trace `INVALID_TX_ALREADY_EXISTS` on method exit? If I add it here won't we be logging it twice?

Reply:
We log trace logs it only in the exception bellow, thing is this might later change to reflect other errors then just `INVALID_TX_ALREADY_EXISTS` so why not trace the error type here.

Reply:
Ah I see there is a log already at the point of discovering the bug ok all good.